src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelectorHeader.test.tsx

```diff
@@ -0,0 +1,109 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { Provider } from 'react-redux';
+import { configureStore, combineReducers } from '@reduxjs/toolkit';
+import { render, getNodeText } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import blastFormReducer, {
+  initialState as initialBlastFormState,
+  type BlastFormState
+} from 'src/content/app/tools/blast/state/blast-form/blastFormSlice';
+
+import SpeciesSelectorHeader, {
+  type Props as SpeciesSelectorHeaderProps
+} from './BlastSpeciesSelectorHeader';
+
+const defaultProps: SpeciesSelectorHeaderProps = {
+  compact: false
+};
+
+const renderComponent = (
+  {
+    props,
+    state
+  }: {
+    state?: Partial<BlastFormState>;
+    props?: SpeciesSelectorHeaderProps;
+  } = { state: {} }
+) => {
+  props = Object.assign({}, defaultProps, props);
+  const blastFormState = Object.assign({}, initialBlastFormState, state);
+  const rootReducer = combineReducers({
+    blast: combineReducers({
+      blastForm: blastFormReducer
+    })
+  });
+  const initialState = {
+    blast: { blastForm: blastFormState }
+  };
+
+  const store = configureStore({
+    reducer: rootReducer,
+    preloadedState: initialState
+  });
+
+  const renderResult = render(
+    <Provider store={store}>
+      <SpeciesSelectorHeader {...props} />
+    </Provider>
+  );
+
+  return {
+    ...renderResult,
+    store
+  };
+};
+
+describe('SpeciesSelectorHeader', () => {
+  describe('species counter', () => {
+    it('starts with 0', () => {
+      const { container } = renderComponent();
+      const speciesCounter = container.querySelector('.header .speciesCounter');
+      expect(getNodeText(speciesCounter as HTMLElement)).toBe('0');
+    });
+
+    it('displays the number of added species', () => {
+      const selectedSpecies = ['human', 'mouse'];
+      const { container } = renderComponent({ state: { selectedSpecies } });
+      const speciesCounter = container.querySelector('.header .speciesCounter');
+      expect(getNodeText(speciesCounter as HTMLElement)).toBe(
+        `${Object.keys(selectedSpecies).length}`
```

Comment:
`Object.keys` — 😞

Reply:
Sorry about that, not sure how this got here, i search for everything and replaced a few before committing the change. I hope thats the last one and i didnt miss any. Done.